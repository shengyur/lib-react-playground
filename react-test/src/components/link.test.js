import React from "react";
import { shallow } from "enzyme";
import Link from "./link";
import renderer from "react-test-renderer";

const props = {
  children: "click this link!",
  page: "https://www.baidu.com"
};

describe("<Link>", () => {
  it("renders correctly", () => {
    const component = renderer.create(<Link {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("hover with correct color", () => {
    const wrapper = shallow(<Link {...props} />);
    expect(wrapper.state("class")).toBe("normal");
    wrapper.simulate("mouseenter");
    expect(wrapper.state("class")).toBe("hovered");
  });
});
