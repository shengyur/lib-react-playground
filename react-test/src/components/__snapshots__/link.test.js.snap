// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> renders correctly 1`] = `
<a
  className="normal"
  href="https://www.baidu.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="__blank"
>
  click this link!
</a>
`;

exports[`<Link> renders correctly 2`] = `
<a
  className="hovered"
  href="https://www.baidu.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="__blank"
>
  click this link!
</a>
`;

exports[`<Link> renders correctly 3`] = `
<a
  className="normal"
  href="https://www.baidu.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="__blank"
>
  click this link!
</a>
`;
